docs/dataservice-cache-example.md

```diff
@@ -0,0 +1,195 @@
+   ```
+
+## Build and Run on Linux
+
+To build and run the example project on Linux:
+
+1. Enable examples of the `CMake` targets.
+
+   ```bash
+   mkdir build && cd build
+   cmake -DOLP_SDK_BUILD_EXAMPLES=ON ..
+   ```
+
+2. In the **build** folder, build the example project.
+
+   ```bash
+   cmake --build . --target dataservice-example
+   ```
+
+3. Execute the example project.
+
+   ```bash
+   ./examples/dataservice-write/dataservice-example --example cache --key_id "here.access.key.id" --key_secret "here.access.key.secret"
+   ```
+
+4. To show help. 
+   ```bash
+   ./examples/dataservice-write/dataservice-example --help
+   ```
+
+After building and running the example project, you see the following information:
+
+```bash
+[INFO] protected-cache-example - Mutable cache path is "none"
+[INFO] protected-cache-example - Protected cache path is "/tmp/cata.log_client_example/cache"
+[INFO] ThreadPoolTaskScheduler - Starting thread 'OLPSDKPOOL_0'
+[INFO] CatalogRepository - cache catalog '@0^0' found!
+[INFO] PartitionsCacheRepository - Get 'hrn:here:data::olp-here-test:edge-example-catalog::versioned-world-layer::1::0::partition'
+[INFO] PartitionsRepository - cache data 'versioned-world-layer[1]@0^0' found!
+[INFO] DataRepository - cache data 'versioned-world-layer[1]@0^0' found!
+[INFO] protected-cache-example - Request partition data - Success, data size - 3375
+```
+
+## How it works
+
+### <a name="authenticate-to-here-olp-using-client-credentials"></a>Authenticate to HERE OLP Using Client Credentials
+
+To authenticate with the Open Location Platform (OLP), you must get platform credentials that contain the access key ID and access key secret.
+
+To authenticate using client credentials:
+
+1. Get your platform credentials. For instructions, see the [Get Credentials](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) section in the Terms and Permissions User Guide.
+
+   You get the `credentials.properties` file.
+
+2. Initialize the authentification settings using the **here.access.key.іd** and **here.access.key.secret** from the `credentials.properties` file as `access_key.id` and `access_key.secret` respectively.
+
+   ```cpp
+   olp::authentication::Settings  settings({access_key.id, access_key.secret});
+   settings.task_scheduler = task_scheduler;
+   settings.network_request_handler = http_client;
+   ```
+
+3. To get the OAuth 2.0 token from OLP, set up the `AuthenticationSettings` object with a default token provider.
+
+   ```cpp
+   olp::client::AuthenticationSettings auth_settings;
+   auth_settings.provider =
+     olp::authentication::TokenProviderDefault(std::move(settings));
+   ```
+
+You can use the `TokenProvider` object to create the `OlpClientSettings` object. For more information, see [Create OlpClientSettings](#create-olpclientsettings).
+
+### <a name="create-olpclientsettings"></a>Create `OlpClientSettings`
```

Comment:
```suggestion
### Create `OlpClientSettings`
```
AFAIK, there's no need to add an extra anchor as `[Take me there](#take-me-there)` automatically anchors the header `Take me there` and the given link.